plugins/dashboards/api/api.js

```diff
@@ -1093,30 +1093,53 @@ plugins.setConfigs("dashboards", {
                 filterCond.owner_id = memberId;
             }
 
-            common.db.collection("dashboards").findOne({_id: common.db.ObjectID(dashboardId)}, function(err, dashboard) {
+            common.db.collection("dashboards").findOne({_id: common.db.ObjectID(dashboardId)}, async function(err, dashboard) {
                 if (err || !dashboard) {
                     common.returnMessage(params, 400, "Dashboard with the given id doesn't exist");
                 }
                 else {
-                    hasViewAccessToDashboard(params.member, dashboard, function(er, status) {
+                    hasViewAccessToDashboard(params.member, dashboard, async function(er, status) {
                         if (er || !status) {
                             return common.returnOutput(params, {error: true, dashboard_access_denied: true});
                         }
 
-                        common.db.collection("dashboards").remove(
-                            filterCond,
-                            function(error, result) {
-                                if (!error && result) {
-                                    if (result && result.result && result.result.n === 1) {
-                                        plugins.dispatch("/systemlogs", {params: params, action: "dashboard_deleted", data: dashboard});
+                        try {
+                            // Remove the dashboard
+                            const result = await common.db.collection("dashboards").deleteOne(filterCond);
```

Comment:
Hello,
If we delete the `dashboards` too early, we lose the opportunity to implement a retry mechanism. However, if we perform the deletion of `dashboards` at a later stage and some cleanup fails, we still have the option to retry.